### PR TITLE
Fixes for issues spotted by coverity (and some cleanups)

### PR DESCRIPTION
**Pull request by keszybz**

### CODE REVIEW

File: src/test/test-process-util.c [hunk 1]
```diff
@@ -425,7 +425,7 @@ static void test_rename_process_multi(void) {
         /* child */
         test_rename_process_now("one", 1);
         test_rename_process_now("more", 0); /* longer than "one", hence truncated */
-        setresuid(99, 99, 99);
+        assert_se(setresuid(99, 99, 99) == 0);
```
Comment (yuwata) [inline src/test/test-process-util.c]: This breaks the test when it is invoked by a non-privileged user.
```
$ ./test-process-util
PID1 comm: 'systemd'
PID1 cmdline: '/usr/lib/systemd/systemd --switched-root --system --deserialize 32'
PID1 cmdline truncated to 8: '/usr...'
PID1 cmdline truncated to 1: ''
PID1 PPID: 0
PID1 exe: 'n/a'
PID1 UID: 0
PID1 GID: 0
PID1 strlen(environ): -13
Failed to read $container of PID 1, ignoring: Permission denied
Found container virtualization none.
PID1 $PATH: 'n/a'
PID6553 comm: 'test-process-ut'
PID6553 cmdline: './test-process-util'
PID6553 cmdline truncated to 8: './te...'
PID6553 cmdline truncated to 1: ''
PID6553 PPID: 1799
PID6553 exe: '/var/build/test-process-util'
PID6553 UID: 1000
PID6553 GID: 1000
PID6553 strlen(environ): 3454
PID6553 $PATH: '/usr/lib64/ccache:/usr/local/bin:/usr/bin:/bin:/usr/local/sbin:/usr/sbin:/home/watanabe/bin:/usr/local/texlive/2017/bin/x86_64-linux'
Skipping PR_SET_MM, as we don't have privileges.
comm = <foo>
cmdline = <foo> (not verified)
Skipping PR_SET_MM, as we don't have privileges.
comm = <this is a reall>
cmdline = <this is a really re> (not verified)
Skipping PR_SET_MM, as we don't have privileges.
comm = <1234567>
cmdline = <1234567> (not verified)
Skipping PR_SET_MM, as we don't have privileges.
comm = <one>
cmdline = <one> (not verified)
Skipping PR_SET_MM, as we don't have privileges.
comm = <more>
cmdline = <mor> (not verified)
Assertion 'setresuid(99, 99, 99) == 0' failed at ../../home/watanabe/git/systemd/src/test/test-process-util.c:428, function test_rename_process_multi(). Aborting.
Assertion 'si.si_code == CLD_EXITED' failed at ../../home/watanabe/git/systemd/src/test/test-process-util.c:419, function test_rename_process_multi(). Aborting.
```

File: src/network/networkd-routing-policy-rule.c [hunk 2]
```diff
@@ -857,24 +857,93 @@ static int routing_policy_rule_read_full_file(char *state_file, char **ret) {
         return size;
 }
 
-int routing_policy_rule_load(Manager *m) {
+int routing_policy_serialize_rules(Set *rules, FILE *f) {
+        RoutingPolicyRule *rule = NULL;
+        Iterator i;
+        int r;
+
```
Comment (yuwata) [inline src/network/networkd-routing-policy-rule.c:864]: It may not be necessary. But I'd like to add `assert(f);` here.

**Comment by keszybz** [comment]
Thanks for the quick review. Updated as suggested. Let's see what CI says.

**Comment by keszybz** [comment]
`191/251 test-resolve                            TIMEOUT 30.03 s`

**Comment by yuwata** [comment]
lgtm.

**Comment by poettering** [comment]
lgtm

**Comment by keszybz** [comment]
I have a bunch of freeing cleanups on top. I'll wait with submitting until this is merged though.

**Comment by yuwata** [comment]
Fedora CI failed with test-resolve timeout, unrelated.

**Comment by poettering** [comment]
OK, the test-resolve thing is fixed already in git master. Since everything else passed will merge

**Pull request merged**